### PR TITLE
Update description of release process to include options for next RCs

### PR DESCRIPTION
**Pull request by potiuk**
Often we excluded individual providers from release and we have two options how to handle them - either release them on their own with an accelerated vote, or include the providers in the next wave of providers.

This PR improves description of our process and email template to explain it better and guide the release manager on what to do.

Also improved notification message

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: dev/README_RELEASE_PROVIDER_PACKAGES.md [hunk 1]
```diff
@@ -992,36 +992,55 @@ that the Airflow works as you expected.
...
 
+[optional] The providers PROVIDER, PROVIDER have been excluded from the release.
+This is due to REASON HERE.
+The next RC candidates for those providers will be released [in the next wave
+of providers] or [as an ad-hoc release on their own with accelerated vote period].
 
-Vote thread:
-https://lists.apache.org/thread.html/736404ca3d2b2143b296d0910630b9bd0f8b56a0c54e3a05f4c8b5fe@%3Cdev.airflow.apache.org%3E
+2 "+1" non-binding votes received:
+
```
Comment (hussein-awala) [inline dev/README_RELEASE_PROVIDER_PACKAGES.md]:
```suggestion
```

File: dev/README_RELEASE_PROVIDER_PACKAGES.md [hunk 2]
```diff
@@ -992,36 +992,55 @@ that the Airflow works as you expected.
...
+- FIRST LAST
+- FIRST LAST
```
Comment (hussein-awala) [inline dev/README_RELEASE_PROVIDER_PACKAGES.md]: Maybe you should move this to below the binding votes.

File: dev/README_RELEASE_PROVIDER_PACKAGES.md (hunk 1 above)
Comment (potiuk) [reply dev/README_RELEASE_PROVIDER_PACKAGES.md]: Yep.

File: dev/README_RELEASE_PROVIDER_PACKAGES.md (hunk 2 above)
Comment (potiuk) [reply dev/README_RELEASE_PROVIDER_PACKAGES.md]: Yes.

**Comment by potiuk** [comment]
Also improved notification message to always include list of packages as "appendix" - previously we had some conditional list there - in order to keep the message short for long list of packages, but I think it makes sense to add the links as appendix instead - most of our users will want to see the direct links to PyPI.

File: dev/README_RELEASE_PROVIDER_PACKAGES.md [hunk 3]
```diff
@@ -992,36 +992,55 @@ that the Airflow works as you expected.
...
 
 3 "+1" binding votes received:
-- Jarek Potiuk  (binding)
-- Kaxil Naik (binding)
-- Tomasz Urbaszek (binding)
+- FIRST LAST NAME (binding)
+- FIRST LAST NAME (binding)
+- FIRST LAST NAME (binding)
+
+2 "+1" non-binding votes received:
```
Comment (hussein-awala) [inline dev/README_RELEASE_PROVIDER_PACKAGES.md]:
```suggestion
```

**Review by hussein-awala — APPROVED** [review]

**Pull request merged**